### PR TITLE
Removed the use of deprecated violation methods

### PR DESCRIPTION
**Pull request by massimo-ferraro**

### CODE REVIEW

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java [hunk 1]
```diff
@@ -135,10 +130,10 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
...
                     String initialStateId = StateManager.INITIAL_STATE_ID;
                     network.getStateManager().setWorkingState(initialStateId);
-                    if ( violatedBranch instanceof Line )
-                        branch = network.getLine(violatedBranch.getId());
-                    else if ( violatedBranch instanceof TwoWindingsTransformer )
-                        branch = network.getTwoWindingsTransformer(violatedBranch.getId());
+                    branch = network.getLine(violation.getSubjectId());
+                    if ( branch == null ) {
+                        branch = network.getTwoWindingsTransformer(violation.getSubjectId());
+                    }
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: Branch has already been set outside the "if" block and these lines seem to be useless?

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java [hunk 2]
```diff
@@ -149,7 +144,7 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
                     } else {
                         LOGGER.warn("State {}: cannot change current limit of branch {}: no branch with this id in the network", 
                                 initialStateId, 
-                                violatedBranch.getId());
+                                violation.getSubjectId());
                     }
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: The else block is unreachable?

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java [hunk 3]
```diff
@@ -163,10 +158,10 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
...
                     String initialStateId = StateManager.INITIAL_STATE_ID;
                     network.getStateManager().setWorkingState(initialStateId);
-                    if ( violatedBranch instanceof Line )
-                        branch = network.getLine(violatedBranch.getId());
-                    else if ( violatedBranch instanceof TwoWindingsTransformer )
-                        branch = network.getTwoWindingsTransformer(violatedBranch.getId());
+                    branch = network.getLine(violation.getSubjectId());
+                    if ( branch == null ) {
+                        branch = network.getTwoWindingsTransformer(violation.getSubjectId());
+                    }
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: Same remarks

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java [hunk 4]
```diff
@@ -177,21 +172,20 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
                     } else {
                         LOGGER.warn("State {}: cannot change current limit of branch {}: no branch with this id in the network", 
                                 initialStateId, 
-                                violatedBranch.getId());
+                                violation.getSubjectId());
                     }
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: The else block is unreachable?

File: modules/src/test/java/eu/itesla_project/modules/constraints/ConstraintsModifierTestUtils.java [hunk 5]
```diff
@@ -100,12 +100,9 @@ public static Network getNetwork() {
 
     public static List<LimitViolation> getViolations(Network network) {
```
Comment (mathbagu) [inline modules/src/test/java/eu/itesla_project/modules/constraints/ConstraintsModifierTestUtils.java]: Network is not needed anymore

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java (hunk 1 above)
Comment (massimo-ferraro) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: Here the code work on a different network state

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java (hunk 3 above)
Comment (massimo-ferraro) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: Here the code works on a different network state

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java (hunk 2 above)
Comment (massimo-ferraro) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: Here the code works on a different network state, that could not have the branch

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java (hunk 4 above)
Comment (massimo-ferraro) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: Here the code works on a different network state, that could not have the branch

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java [hunk 6]
```diff
@@ -133,24 +126,17 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
...
-                    else if ( violatedBranch instanceof TwoWindingsTransformer )
-                        branch = network.getTwoWindingsTransformer(violatedBranch.getId());
-                    if ( branch != null ) {
-                        LOGGER.debug("State {}: changing current limit 1 of branch {}: {} -> {}", 
-                                initialStateId, 
+                    network.getStateManager().setWorkingState(StateManager.INITIAL_STATE_ID);
+                    branch = network.getLine(violation.getSubjectId());
+                    if ( branch == null ) {
+                        branch = network.getTwoWindingsTransformer(violation.getSubjectId());
+                    }
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: This block is useless.

File: modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java [hunk 7]
```diff
@@ -161,37 +147,29 @@ else if ( violatedBranch instanceof TwoWindingsTransformer )
...
-                    else if ( violatedBranch instanceof TwoWindingsTransformer )
-                        branch = network.getTwoWindingsTransformer(violatedBranch.getId());
-                    if ( branch != null ) {
-                        LOGGER.debug("State {}: changing current limit 2 of branch {}: {} -> {}", 
-                                initialStateId, 
+                    network.getStateManager().setWorkingState(StateManager.INITIAL_STATE_ID);
+                    branch = network.getLine(violation.getSubjectId());
+                    if ( branch == null ) {
+                        branch = network.getTwoWindingsTransformer(violation.getSubjectId());
+                    }
```
Comment (mathbagu) [inline modules/src/main/java/eu/itesla_project/modules/constraints/ConstraintsModifier.java]: This block is useless

**Pull request merged**